cirq/linalg/combinators.py

```diff
@@ -112,6 +114,31 @@ def dot(*values: Union[float, complex, np.ndarray]
     return np.linalg.multi_dot(values)
 
 
+def dot(*values: Union[float, complex, np.ndarray]
+       ) -> Union[float, complex, np.ndarray]:
+    """Computes the dot/matrix product of a sequence of values.
+
+    Performs the computation in serial order without regard to the matrix
+    sizes.  If you are using this for matrices of large and differing sizes,
+    consider using multi_dot for better performance
+
+    Args:
+        *values: The values to combine with the dot/matrix product.
+
+    Returns:
+        The resulting value or matrix.
+    """
+
+    if len(values) == 1:
+        if isinstance(values[0], np.ndarray):
+            return np.array(values[0])
+        return values[0]
```

Comment:
I think you meant `if not isintance...`.

You can juse use `return np.asarray(values[0])` here.

Reply:
I wouldn't do that. The caller may assume that the output will not alias with the input.

Reply:
I don't think this is correct.  I think the original intent was to return a copy of the array.

Reply:
Then why `return values[0]`?

Reply:
According to the mypy type annotation, if it's not an numpy array, it's a float or complex which are immutable.